### PR TITLE
Update Electron, Typescript example

### PR DESCRIPTION
**Pull request by laiso**
It seems like the error occurs with `next export` when building the current example.

```
$ cd examples/with-electron-typescript
$ yarn && yarn build

Error occurred prerendering page "/detail". Read more: https://err.sh/next.js/prerender-error
Error: Error for page /detail: pages with `getServerSideProps` can not be exported. See more info here: https://err.sh/next.js/gssp-export
Error: Export encountered errors on following paths:
        /detail
    at exportApp (/Users/kstg/src/github.com/zeit/next.js/examples/with-electron-typescript/node_modules/next/dist/export/index.js:22:1166)
error Command failed with exit code 1.   
```

I changed routes to statically resolve pages generating from `pages/detail.tsx` and added `getStaticPaths` and `getStaticProps`. so then it will export as follows

```
yarn run v1.22.4
$ next build renderer && next export renderer
Creating an optimized production build...

Compiled successfully.

Automatically optimizing pages...

Page                            Size     First Load JS
┌ ○ /                           336 B          62.6 kB
├ ○ /404                        2.54 kB        61.6 kB
├ ○ /about                      338 B          62.6 kB
├ ● /detail/[id]                440 B          62.7 kB
├   ├ /detail/101
├   ├ /detail/102
├   ├ /detail/103
├   └ /detail/104
└ ● /initial-props              530 B          62.8 kB
+ First Load JS shared by all   59.1 kB
  ├ static/pages/_app.js        983 B
  ├ chunks/commons.9facbe.js    10.9 kB
  ├ chunks/framework.b76773.js  40.3 kB
  ├ runtime/main.800476.js      6.22 kB
  └ runtime/webpack.c21266.js   746 B

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)

> using build directory: /Users/kstg/src/github.com/zeit/next.js/examples/with-electron-typescript/renderer/.next
  copying "static build" directory
> No "exportPathMap" found in "next.config.js". Generating map from "./pages"
  launching 7 workers
Exporting (0/4)
Exporting (1/4)
Exporting (2/4)
Exporting (3/4)
Exporting (4/4)

Export successful
Done in 11.47s.
```

### CODE REVIEW

File: examples/with-electron-typescript/renderer/components/ListItem.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ type Props = {
 }
 
 const ListItem = ({ data }: Props) => (
-  <Link href={`/detail?id=${data.id}`}>
+  <Link href={`/detail/${data.id}`}>
```
Comment (lfades) [inline examples/with-electron-typescript/renderer/components/ListItem.tsx:11]:
```suggestion
  <Link href="/detail/[id]" as={`/detail/${data.id}`}>
```

File: examples/with-electron-typescript/renderer/pages/detail/[id].tsx [hunk 2]
```diff
@@ -30,10 +30,17 @@ const InitialPropsDetail = ({ item, errors }: Props) => {
...
+  const items: User[] = await findAll()
+  const paths = items.map((item) => `/detail/${item.id}`)
+  return { paths, fallback: false }
+}
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const id = context.params?.id
+  if (!id) {
+    throw new Error('Invalid page id')
+  }
```
Comment (lfades) [inline examples/with-electron-typescript/renderer/pages/detail/[id].tsx:43]: No need to validate the id here.

```suggestion
export const getStaticProps: GetStaticProps = async ({ params }) => {
  const { id } = params
```

**Review by lfades — APPROVED** [review]
@laiso Good work. Thank you!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| buildDuration | 9.7s | 9.9s | ⚠️ +266ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.714 | 1.739 | ⚠️ +0.03 |
| / avg req/sec | 1458.84 | 1437.6 | ⚠️ -21.24 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 0.951 | 0.938 | -0.01 |
| /error-in-render avg req/sec | 2629.81 | 2664.99 | +35.18 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-c5a676d..82de.js gzip | 6.75 kB | 6.75 kB | ✓ |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 87.6 kB | 87.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-079bffd..dule.js gzip | 5.83 kB | 5.83 kB | ✓ |
| polyfills-05..dule.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 82.6 kB | 82.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| index.html gzip | 947 B | 947 B | ✓ |
| link.html gzip | 951 B | 951 B | ✓ |
| withRouter.html gzip | 937 B | 937 B | ✓ |
| Overall change | 2.83 kB | 2.83 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| buildDuration | 10.9s | 10.7s | -207ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-c5a676d..82de.js gzip | 6.75 kB | 6.75 kB | ✓ |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 87.6 kB | 87.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-079bffd..dule.js gzip | 5.83 kB | 5.83 kB | ✓ |
| polyfills-05..dule.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 82.6 kB | 82.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.2 MB | 5.2 MB | ✓ |

</details>

</details>
Commit: d243d8a95a539e9c7664a84bc6a6aaf4eae642ec

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.5s | ⚠️ +213ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.425 | 2.477 | ⚠️ +0.05 |
| / avg req/sec | 1030.8 | 1009.14 | ⚠️ -21.66 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.461 | 1.552 | ⚠️ +0.09 |
| /error-in-render avg req/sec | 1711.42 | 1610.68 | ⚠️ -100.74 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-c5a676d..82de.js gzip | 6.75 kB | 6.75 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-079bffd..dule.js gzip | 5.83 kB | 5.83 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| index.html gzip | 947 B | 947 B | ✓ |
| link.html gzip | 951 B | 951 B | ✓ |
| withRouter.html gzip | 937 B | 937 B | ✓ |
| Overall change | 2.83 kB | 2.83 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| buildDuration | 15.5s | 15.2s | -349ms |
| nodeModulesSize | 65.5 MB | 65.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..b7a9.js gzip | 10.2 kB | 10.2 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-c5a676d..82de.js gzip | 6.75 kB | 6.75 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 56.8 kB | 56.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.09 kB | 6.09 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-079bffd..dule.js gzip | 5.83 kB | 5.83 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 51.8 kB | 51.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| Overall change | 30.8 kB | 30.8 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-a98d9..5cb7.js gzip | 3.45 kB | 3.45 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.71 kB | 7.71 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-d5979..dule.js gzip | 2.3 kB | 2.3 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.36 kB | 5.36 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | laiso/next.js fix-example-electron-typescript | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.2 MB | 5.2 MB | ✓ |

</details>

</details>
Commit: 4c177aba3f29bf5da1cdbb800c9ea4d3396acba2

**Pull request merged**